### PR TITLE
Feature/sound

### PR DESCRIPTION
**Pull request by notzain**
Works:

* Play music
* Play sound effects
* Loop sound effects

TBD:
* See if sound needs to be on a different thread
* Volume adjustments

### CODE REVIEW

**Comment by FabioGNR** [comment]
👍

File: engine/sound/SDL/SDLSoundManager.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+static std::map<std::string, std::unique_ptr<Mix_Music, void (*)(Mix_Music*)>> m_musicCache;
+
+namespace engine {
+namespace sound {
+    SDLSoundManager::SDLSoundManager()
+    {
+        static bool isInitialized = false;
+
+        if (!isInitialized) {
+            if (Mix_OpenAudio(22050, MIX_DEFAULT_FORMAT, 2, 4096) == -1) {
```
Comment (oblekta) [inline engine/sound/SDL/SDLSoundManager.cpp]: What does 22050 and 4096 mean? Perhaps create two constants for clarity

File: engine/sound/SDL/SDLSoundManager.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    void SDLSoundManager::pause() const
+    {
+        Mix_PausedMusic();
+    }
+
+    void SDLSoundManager::setMusicVolume(const Volume volume)
+    {
+        const int sdl_volume = static_cast<int>(volume.volume * 1.28);
```
Comment (oblekta) [inline engine/sound/SDL/SDLSoundManager.cpp]: Why 1.28?

File: engine/sound/SDL/SDLSoundManager.cpp (hunk 1 above)
Comment (notzain) [reply engine/sound/SDL/SDLSoundManager.cpp]: Please review latest commit.

**Pull request merged**